proposals/0317-sdl-protocol-security-specification.md

```diff
@@ -370,7 +288,117 @@ The following query header is used by the system and the application to send err
+
+The RPC service needs to be started as unencrypted first, then moved to encrypted state by sending another `StartService` request at a later point. Other services can do the same thing to move from unencrypted to encrypted.
+
+Before the encryption of RPC service is enabled (encryption is not available), SDL Core rejects any RPC request with result code `ENCRYPTION_NEEDED` if the RPC needs protection (please see policy updates for which RPC needs protection). SDL Core continues processing an RPC request if the RPC does not need protection. SDL Core sends a notification only if the notification RPC does not need protection.
+
+After the encryption of RPC service is enabled (encryption is available), SDL Core rejects any unencrypted RPC requests with result code `ENCRYPTION_NEEDED` with the unencrypted response if the RPC needs protection. SDL Core continues processing an unencrypted RPC request if the RPC does not need protection and responds with an unencrypted response. SDL Core continues processing an encrypted RPC request if the RPC needs protection and responds with an encrypted response. SDL Core sends an unencrypted notification if the RPC does not need protection. SDL Core sends an encrypted notification if the RPC needs protection. In addition, SDL Core shall continue processing an encrypted RPC request if the RPC does not need protection and responds with an encrypted response.
+
+### 7.1 Authentication
+
+The below diagram shows the sequence of how the TLS handshake exchanges certificates to compute the master secret.
+
+![TLS Handshake activity diagram](../assets/proposals/0317-sdl-protocol-security-specification/tls-handshake.png)
+
+The authentication is done using TLS handshake. The TLS handshake process is defined by TLS and is not part of the SDL protocol. The handshake is designed as a client server communication which is configurable in the system settings. An application should take the role of a server where the system is the client. The client entity will initiate a TLS handshake with the corresponding security manager of the server. The client will do this only if the server was not authenticated before in the current transport connection. According to the TLS handshake process the peer certificate can be omitted for the server but it's required for the client. Certificate peer verification can be enabled/disabled on Core side by changing `VerifyPeer` parameter in the configuration file. From another hand, mobile side library does not require certificate from Core so it can be omitted.
```

Comment:
> mobile side library does not require certificate from Core so it can be omitted

Looks like this statement requires an additional update.

From TLS Handshake point of view - that's absolutely correct. 
However before the handshake Core verifies if it's certificate is actual. If it's missing (or outdated/invalid) then Core initiate PTU to obtain new certificate from Policy Server. And (most important part) Core **does not start handshake** if this certificate can't be obtained. Core in this case fails starting of protected service. (Please double check on a Core's code level).

Hence Core certificate is required for starting of protected service. But is not required for TLS handshake.